### PR TITLE
Feature/documentdb

### PR DESCRIPTION
**Pull request by thomas-maschler**
Document DB module

### CODE REVIEW

File: terraform/vars/terraform-dev.tfvars [hunk 1]
```diff
@@ -1,4 +1,9 @@
...
-dns_prefix               = "aws-dev"
-backups_bucket           = "wri-api-dev-backups"
\ No newline at end of file
+environment                 = "dev"
+ingress_allow_cidr_block    = "0.0.0.0/0"
+dns_prefix                  = "aws-dev"
+backups_bucket              = "wri-api-dev-backups"
+log_retention_period        = 7
+backup_retention_period     = 1
+db_instance_class           = "db.t3.medium"
```
Comment (tiagojsag) [inline terraform/vars/terraform-dev.tfvars]: given we will soon have multiple DBs, I would improve the naming on these variables so it's clearer

File: terraform/main.tf [hunk 2]
```diff
@@ -199,6 +200,22 @@ module "gfw-pro-node-group" {
...
+  private_subnet_ids              = [module.vpc.private_subnets[0].id, module.vpc.private_subnets[1].id, module.vpc.private_subnets[3].id]
+  project                         = local.project
+  backup_retention_period         = var.backup_retention_period
+  instance_class                  = var.db_instance_class
+  cluster_size                    = var.db_instance_count
+  master_password                 = var.db_password
+  master_username                 = "wri" # superuser, create app specific users at project level
+  tags                            = local.tags
+  vpc_id                          = module.vpc.id
+  engine_version                  = "3.6.0"
```
Comment (tiagojsag) [inline terraform/main.tf:215]: I'm assuming this is the latest engine version that's compatible with MongoDB 3.6?

File: terraform/main.tf [hunk 3]
```diff
@@ -199,6 +200,22 @@ module "gfw-pro-node-group" {
   }
 }
 
+module "documentdb" {
+  source                          = "./modules/document_db"
+  log_retention_period            = var.log_retention_period
+  private_subnet_ids              = [module.vpc.private_subnets[0].id, module.vpc.private_subnets[1].id, module.vpc.private_subnets[3].id]
+  project                         = local.project
+  backup_retention_period         = var.backup_retention_period
```
Comment (tiagojsag) [inline terraform/main.tf:207]: This is something I have pending to ask the AWS ES specialist, and probably we'll have the same question for the other DBs: What happens to backups if, for example, the DocumentDB cluster fails in an irrecoverable way (it has happened with EKS on dev) or if we accidentally remove the DocumentDB cluster? I know it's an unlikely scenario, but it still feels like we're putting all of our eggs in a very tightly coupled basket, and drifting away from the 3-2-1 backup rule.

File: terraform/modules/document_db/main.tf [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+}
+
+
+####################
+# Secret Manager
+####################
+
+resource "aws_secretsmanager_secret" "documentdb" {
+  description = "Connection string for DocumentDB cluster"
+  name        = "${var.project}-docuementdb-secret"
```
Comment (tiagojsag) [inline terraform/modules/document_db/main.tf]: typo

**Review by tiagojsag — COMMENTED** [review]
overal LGTM, just a couple of minor notes here and there

File: terraform/main.tf (hunk 3 above)
Comment (thomas-maschler) [reply terraform/main.tf:207]: keeping the current backup strategy in place (full daily, weekly and monthly dumps) plus AWS cluster backup for now

File: terraform/main.tf (hunk 2 above)
Comment (thomas-maschler) [reply terraform/main.tf:215]: yes

**Review by ikas — APPROVED** [review]
LGTM, approving the PR :+1: 

However, I have some thoughts on the changes made to the backup scripts (cc @thomas-maschler @tiagojsag). 

The existing backups for Mongo CT DB **do not backup the `statistics` collection**. This is because this particular collection is huge and it would increase (nay, double) the size of the backups, when in fact, there's no real need for keeping backups of these statistics.

A while ago (when we merged MongoCT + Mongo apps into a single instance), we changed the backup scripts, from having a single backup script for all DBs to having a single backup script per DB being backed up. This was done because there's a restriction in `mongodump` that only allows excluding DBs from the backup when backing up multiple DBs (not one particular table inside a particular DB, which would have been what we need).

I think these changes will now include the `statistics` table in the backups, so it's very likely that we see an increase in the size of the backups - not a blocker IMO, but something to be taken into account!

**Comment by thomas-maschler** [comment]
Hey @ikas , 

I saw that option for the Control Tower backup and addressed it here
https://github.com/resource-watch/api-infrastructure/blob/feature/documentdb/k8s-aws/backups/entrypoint.sh#L57

the statistics table should not be included in the backup

**Comment by ikas** [comment]
~~The problem is that `exclude-collections` is not taken into consideration when backing up multiple databases, only when backing up a single DB ([you can see in the comments that DB name is required for exclude collections](https://github.com/resource-watch/api-infrastructure/blob/117ff6e720cba8bd84c680798f8ae971f7adf55e/k8s-aws/backups/automongobackup.sh#L52)). Hence why we migrated to one file per DB.~~

~~In any case, as I said, I don't see this as a blocker for merging the PR. So, I'd suggest you go ahead and merge the PR, then test the backups in the dev/staging cluster and go from there. Might not such a bad thing if we include the `statistics` collection in the dump after all 😅~~

**Comment by ikas** [comment]
Nevermind what I said above, just realized you are doing a single backup for the CT database. Perfect 👌 then I'm all good!

**Pull request merged**